src/SortMsg.chpl

```diff
@@ -29,157 +30,82 @@ module SortMsg
     }
 
     /* sort takes pdarray and returns a sorted copy of the array */
-    @arkouda.registerND
-    proc sortMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-      param pn = Reflection.getRoutineName();
-      const algoName = msgArgs.getValueOf("alg"),
-            name = msgArgs.getValueOf("array"),
-            axis = msgArgs.get("axis").getIntValue(),
-            rname = st.nextName();
-
-      var algorithm: SortingAlgorithm = defaultSortAlgorithm;
-      if algoName != "" {
-        try {
-          algorithm = algoName: SortingAlgorithm;
-        } catch {
-          throw getErrorWithContext(
-            msg="Unrecognized sorting algorithm: %s".format(algoName),
-            lineNumber=getLineNumber(),
-            pn,
-            moduleName=getModuleName(),
-            errorClass="NotImplementedError"
-          );
-        }
+    @arkouda.registerCommand
+    proc sort(array: [?d] ?t, alg: string, axis: int): [d] t throws 
+    where ((t == real) || (t == int) || (t == uint(64))) && (d.rank == 1) {
+
+      var algorithm: SortingAlgorithm = ArgSortMsg.getSortingAlgorithm(alg);
+      const itemsize = dtypeSize(whichDtype(t));
+      overMemLimit(radixSortLSD_keys_memEst(d.size, itemsize));
+
+      if algorithm == SortingAlgorithm.TwoArrayRadixSort {
+        var sorted = makeDistArray(array);
+        ArgSortMsg.dynamicTwoArrayRadixSort(sorted, comparator=myDefaultComparator);
+        return sorted;
+      } else {
+        var sorted = radixSortLSD_keys(array);
+        return sorted;
```

Comment:
I thought these had to return a message tuple, so to insert an array into the symbol table and return a success you would do like
```chapel 
return st.insert(new shared SymEntry(sorted));
```

or `MsgTuple.error` for the errors